### PR TITLE
Switch to rippled xrpl-core library:

### PR DESCRIPTION
**Pull request by mellery451**
Incorporate rippled 1.1.0, which now provides xrpl_core library with
the same functionality as this library previous did. Vastly simplify
the CMake file here to build example signing application against the
core library via subdirectory inclusion.

### CODE REVIEW

File: README.md
Comment (ximinez) [inline README.md:97]: The README needs to be updated for Windows builds to describe the patch that you made to appveyor.yml, and to update the target to `Visual Studio 15 2017 Win64".
```
       (Get-Content extras/rippled/CMakeLists.txt).replace('exclude_if_included (rocksdb)', '##') | Set-Content extras/rippled/CMakeLists.txt
```

**Comment by ximinez** [comment]
Wow. This almost obsoletes this entire project.

**Comment by mellery451** [comment]
yeah - you're right. With this change, the project essentially serves as a demo of how to use `Ripple:xrpl_core`, so might be worth a rename eventually (?)

**Review by ximinez — CHANGES_REQUESTED** [review]
Please add `.nih_c` to `.gitignore`

File: README.md
Comment (mellery451) [reply README.md:97]: good point - or we just put this on hold until a fix lands in `develop` (or `master`) and switch to it once the dirty hack is no longer needed. I'm open to either approach, although I hate asking users to hack around bugs.

Comment (ximinez) [reply README.md:97]: I'd rather wait than ship broken code....

Could the patch be scripted into CMake, the removed when the fix gets merged?

**Review by ximinez — APPROVED** [review]
YAAASSSSS!!!!

File: CMakeLists.txt
Comment (seelabs) [inline CMakeLists.txt:24]: I'd expect a new beta soon. What do you think about holding up this PR so we can remove this hack?

**Review by seelabs — COMMENTED** [review]
:+1: Though it would be nice to get rid of the hack to patch rippled.

File: CMakeLists.txt
Comment (mellery451) [reply CMakeLists.txt:24]: I think the submodule points to `master` (as it should), which is more like a 3 month wait...which I think is still fine if we want to hold it.

Comment (ximinez) [reply CMakeLists.txt:24]: I vote to merge with the hack:
1. The patch only applies to one environment (Windows)
2. Once the `rippled` fix is merged to master, the affected line will (presumably) be fixed to be in a block that is only relevant in non-Windows. (And if it doesn't get merged, then the hack stays relevant.)
3. Because the patch modifies a git tracked file, I think anybody who does a Windows built is likely to notice as soon as they run `git status` or even sooner if they use one of the various git aware shell prompts.

Comment (seelabs) [reply CMakeLists.txt:24]: I'm fine with merging with the hack.

**Pull request merged**
